### ISSUE TITLE
dynamic deephit plugin does not work on a simple dataset

### ISSUE THREAD

**Issue by WenjuanW** [vanderschaarlab/temporai#70]
Hi,

I created a simple dataset according to the example in data format tutorial.

```
time_series_df = pd.DataFrame(
    {
        "sample_idx": ["sample_0", "sample_0", "sample_0", "sample_1", "sample_1", "sample_1", "sample_2", "sample_2", "sample_2"],
        "time_idx": [1, 2, 3, 1, 2, 3, 1, 2, 3],
        "t_feat_0": [11, 12, 13, 14, 21, 22, 31, 28, 26],
        "t_feat_1": [1.1, 1.2, 1.3, 1, 2.1, 2.2, 3.1, 2.3, 2.0],
    }
)

# Set the 2-level index:
time_series_df.set_index(keys=["sample_idx", "time_idx"], drop=True, inplace=True)

# Create a static data dataframe.
static_df = pd.DataFrame(
    {
        "s_feat_0": [100, 200, 300],
        "s_feat_1": [-1.1, -1, -1.3],
    },
    index=["sample_0", "sample_1", "sample_2"],
)

# Create an event dataframe.

event_df = pd.DataFrame(
    {
        "e_feat_0": [(10, True), (12, False), (13, True)],
    },
    index=["sample_0", "sample_1", "sample_2"],
)

# Create a dataset of time-to-event analysis task:
data = TimeToEventAnalysisDataset(
    time_series=time_series_df,
    static=static_df,
    targets=event_df,
)
```
But the dynamic_deephit plugin does not work on the above dataset with the following errors.

![Screenshot 2023-07-26 at 15 00 55](https://github.com/vanderschaarlab/temporai/assets/25642178/ffb498f8-2986-4f9d-8755-5f26025fc89a)

It seems that the dataset was not fit-ready. I checked that I have all the components for time to event dataset. Could you help with this please?

Thanks very much,
Wenjuan

**Comment by DrShushen** [comment]
Will be looking into this shortly, thank you for your patience!

**Comment by DrShushen** [comment]
The issue has now been fixed in the latest `main` branch of TemporAI.
Please install with
```bash
pip install git+https://github.com/vanderschaarlab/temporai.git
```
until `v0.0.3` is released.

### Cause
The dynamic deephit model creates a validation set at training time, in order to do early stopping. The validation set fraction default is `0.1` (`* num_samples`). Hence this example with only 3 samples would lead to a validation set with 0 samples, causing the error.

### Fix
Fixed in #74. The above example with
```
model = plugin_loader.get("time_to_event.dynamic_deephit", n_iter=50)
model.fit(data)
```
will now raise an error with a clear explanation:
`RuntimeError: Not enough samples to create a validation set, please increase `val_size` or use a larger dataset. `val_size` was 0 and total number of samples in dataset was 3.`

It is now possible to pass the `val_size` parameter when creating the plugin. So, the above examples can be made to work by passing a larger `val_size`, e.g.:
```python
import pandas as pd

from tempor.data.dataset import TimeToEventAnalysisDataset
from tempor.plugins import plugin_loader

time_series_df = pd.DataFrame(
    {
        "sample_idx": ["sample_0", "sample_0", "sample_0", "sample_1", "sample_1", "sample_1", "sample_2", "sample_2", "sample_2"],
        "time_idx": [1, 2, 3, 1, 2, 3, 1, 2, 3],
        "t_feat_0": [11, 12, 13, 14, 21, 22, 31, 28, 26],
        "t_feat_1": [1.1, 1.2, 1.3, 1, 2.1, 2.2, 3.1, 2.3, 2.0],
    }
)

# Set the 2-level index:
time_series_df.set_index(keys=["sample_idx", "time_idx"], drop=True, inplace=True)

# Create a static data dataframe.
static_df = pd.DataFrame(
    {
        "s_feat_0": [100, 200, 300],
        "s_feat_1": [-1.1, -1, -1.3],
    },
    index=["sample_0", "sample_1", "sample_2"],
)

# Create an event dataframe.

event_df = pd.DataFrame(
    {
        "e_feat_0": [(10, True), (12, False), (13, True)],
    },
    index=["sample_0", "sample_1", "sample_2"],
)

# Create a dataset of time-to-event analysis task:
data = TimeToEventAnalysisDataset(
    time_series=time_series_df,
    static=static_df,
    targets=event_df,
)

model = plugin_loader.get(
    "time_to_event.dynamic_deephit",
    n_iter=50,
    val_size=0.4,  # NOTE.
)
model.fit(data)
```
Note that a `RuntimeWarning` about small validation set size will be shown (if fewer than 10 samples).
